### PR TITLE
Bugfix: cant create order from admin backend

### PR DESCRIPTION
**Pull request by sreichel**
### Related Pull Requests

1. See OpenMage/magento-lts#4266

### Fixed Issues (if relevant)

1. Fixes OpenMage/magento-lts#4347

### Manual testing scenarios (*)

1. create order from admin backend
2. ...

@Hanmac  please double-check. Maybe something is wrong at my side.

### CODE REVIEW

**Review by Hanmac — APPROVED** [review]

**Comment by fballiano** [comment]
It would be better to rever that pr completely!

**Comment by fballiano** [comment]
because it's broken and you all merged it with zero testing!!!

**Comment by Hanmac** [comment]
> because it's broken and you all merged it with zero testing!!!

It got reviewed over multiple weeks and no one noticed

**Comment by fballiano** [comment]
none noticed that you can't view or create an order because nobody tested it!

**Comment by fballiano** [comment]
I ported it because I trusted this repo, you're telling I'm wrong?

I'll probably stop porting anything if this is the quality since the day I stop managing this repo

**Comment by fballiano** [comment]
> AND ... you merged it too - w/o further reviews ;)

what are you winking at precisely?

**Comment by sreichel** [comment]
> I'll probably stop porting anything if this is the quality since the day I stop managing this repo

This would give you more time to work on OM. :)

A bug was found. A bug was fixed. There was no release in between. Everything is fine. Not?

**Comment by fballiano** [comment]
it's amazing

**Comment by fballiano** [comment]
> > I'll probably stop porting anything if this is the quality since the day I stop managing this repo
> 
> This would give you more time to work on OM. :)

this sentence doesn't make any sense.

**Review by addison74 — DISMISSED** [review]

**Comment by sreichel** [comment]
> this sentence doesn't make any sense.

Instead of complaining about OM you could contribute .... so you had not do deal with nasty bugs in your fork.

**Comment by fballiano** [comment]
> Instead of complaining about OM you could contribute .... so you had not do deal with nasty bugs in your fork.

Don't worry, I will deal with that in other ways, you should care about not destroying this project instead.

File: app/code/core/Mage/Adminhtml/Block/Sales/Order/Create/Giftmessage.php [hunk 1]
```diff
@@ -82,6 +82,6 @@ public function canDisplayGiftmessage(): bool
         }
         /** @var Mage_GiftMessage_Helper_Message $helper */
         $helper = $this->helper('giftmessage/message');
-        return $helper->getIsMessagesAvailable('order', $this->getEntity(), $this->getEntity()->getStoreId());
+        return $helper->getIsMessagesAvailable('main', $this->getQuote(), $this->getStoreId());
```
Comment (kiatng) [inline app/code/core/Mage/Adminhtml/Block/Sales/Order/Create/Giftmessage.php:85]: I think the first param `$type` refers to the type of entity, so
```suggestion
        return $helper->getIsMessagesAvailable('quote', $this->getQuote(), $this->getStoreId());
```

**Comment by kiatng** [comment]
@fballiano We highly value your contributions and feedback, but I noticed that some of your comments might not align with our [Code of Conduct](https://github.com/OpenMage/magento-lts?tab=coc-ov-file).

For example, comments like "because it's broken and you all merged it with zero testing!!!" and "none noticed that you can't view or create an order because nobody tested it!" could be perceived as confrontational and not in the spirit of respectful and constructive communication that we strive for in our community.

Our Code of Conduct encourages using welcoming and inclusive language, being respectful of differing viewpoints, and gracefully accepting constructive criticism. We believe that maintaining a positive and supportive environment is crucial for the success of our project and the well-being of all contributors.

We appreciate your understanding and cooperation in this matter.

**Comment by fballiano** [comment]
@kiatng I think answering "Instead of complaining about OM you could contribute .... so you had not do deal with nasty bugs in your fork." to the biggest contributor, not only by commits on the main branch, the immense number of backports to v19, the immense commits and forward port to next (which by the way has been thrown away), the immense public relation work and PR revisions, is disrespectful. Also, caliing it "your fork" is just a way to diminish other's people work. And this is all without touching the subject of "who" and why the COC "punishments" were canceled without any public statement (actually there were public statements that said the opposite of what happened).

Also, on the subject, do you think there is enough testing lately? hundreds and hundreds of files are modified almost daily, the errors came up after the PR in subject are very important so my doubts are relevant.

File: app/code/core/Mage/Adminhtml/Block/Sales/Order/Create/Giftmessage.php (hunk 1 above)
Comment (Hanmac) [reply app/code/core/Mage/Adminhtml/Block/Sales/Order/Create/Giftmessage.php:85]: @kiatng it was `main` before my MR, so 🤷‍♂️ 

https://github.com/OpenMage/magento-lts/pull/4266/files#diff-7d3121e3c6013546087c47238a88d86e0f15c12cbc344b8e893cbf1e94b19dc3L18

Comment (sreichel) [reply app/code/core/Mage/Adminhtml/Block/Sales/Order/Create/Giftmessage.php:85]: https://github.com/OpenMage/magento-lts/blob/main/app/code/core/Mage/GiftMessage/Helper/Message.php#L95-L147

It seems it doesnt matter if its `main` or `quote`.

@Hanmac can you please check again to not have replaced any of the possible types? (items, item, order_item, or address_item),

Comment (sreichel) [reply app/code/core/Mage/Adminhtml/Block/Sales/Order/Create/Giftmessage.php]: Added tests. LGTM.

**Review by kiatng — APPROVED** [review]

**Review by addison74 — APPROVED** [review]

**Pull request merged**